query-engine/js-connectors/js/neon-js-connector/src/neon.ts

```diff
@@ -18,22 +18,31 @@ const TRANSACTION_ROLLBACK = 'ROLLBACK'
 class PrismaNeon implements Connector, Closeable {
   readonly flavor = 'postgres'
 
-  private client: Client
+  private client?: Client
+  private httpClient?: NeonQueryFunction<false, true>
   private isRunning: boolean = true
   private inTransaction: boolean = false
   private _isHealthy: boolean = true
   private _version: string | undefined = undefined
+  private _httpMode: boolean
```

Comment:
Ideally this could be

```suggestion
  private mode: 'http' | 'ws'?
```

Currently, we're using `_` prefixes only on field names that happen to conflict with method names in `Connector`.

Reply:
(Also, notice how we mark every variable which needs a default value outside the constructor)

Reply:
Deprecated this comment in favor of https://github.com/prisma/prisma-engines/pull/4110/files#r1286118744.